sklearn_numba_dpex/common/random.py

```diff
@@ -158,39 +158,46 @@ def create_xoroshiro128pp_states(n_states, subsequence_start=0, seed=None, devic
     if hasattr(seed, "randint"):
         seed = uint64(seed.randint(0, np.iinfo(np.int64).max - 1))
 
-    seed = uint64(seed)
-    n_states = int64(n_states)
+    init_xoroshiro128pp_states_kernel = _make_init_xoroshiro128pp_states_kernel(
+        n_states, subsequence_start
+    )
 
-    init_const_1 = np.uint64(0)
+    # Initialization is purely sequential so it will be faster on CPU, if a cpu device
+    # is available make sure to use it.
+    if device is None:
+        device = dpctl.SyclDevice()
+    from_cpu_to_device = False
+    if not device.has_aspect_cpu:
+        try:
+            cpu_device = dpctl.SyclDevice("cpu")
+            from_cpu_to_device = True
+        except dpctl.SyclDeviceCreationError:
+            warnings.warn(
+                "No CPU found, falling back random initiatlization to default device."
+            )
```

Comment:
Side-note: This code is also present in KMeans.

Don't we have an utility function for that somewhere?

Reply:
Last time we chose to wait for the third occurrence to chose to factor it. Third occurrence happens in https://github.com/soda-inria/sklearn-numba-dpex/pull/81 + the said refacto.

Reply:
Ah yes!